### PR TITLE
Fix intermittent test failures by not mutating global state

### PR DESCRIPTION
**Pull request by bf4**
e.g.

https://ci.appveyor.com/project/bf4/active-model-serializers/build/415/job/qs8895cvla8fl0rf
 1) Failure:
 ActiveModel::Serializer::Adapter::JsonApi::RelationshipTest#test_relationship_including_data_explicit
 [C:/projects/active-model-serializers/test/adapter/json_api/relationships_test.rb:186]:
 --- expected
 +++ actual
 @@ -1 +1 @@
 -{:data=>{:id=>"1337", :type=>"authors"}, :meta=>{:name=>"Dan Brown"}}
 +{:data=>{:id=>"1337", :type=>"authors"

https://travis-ci.org/rails-api/active_model_serializers/jobs/156678147
  1) Failure:
  ActiveModel::Serializer::Adapter::JsonApi::RelationshipTest#test_relationship_with_everything
  [/home/travis/build/rails-api/active_model_serializers/test/adapter/json_api/relationships_test.rb:200]:
  --- expected
  +++ actual
  @@ -1 +1 @@
  -{:data=>[{:id=>"1337", :type=>"likes"}],
  :links=>{:related=>{:href=>"//example.com/likes/1337",
  :meta=>{:ids=>"1337"}}}, :meta=>{:liked=>true}}
  +{:data=>[{:id=>"1337", :type=>"likes"}],
  :links=>{:related=>{:href=>"//example.com/likes/1337",
  :meta=>{:ids=>"1337"}}}}

### CODE REVIEW

File: test/adapter/json_api/relationships_test.rb [hunk 1]
```diff
@@ -103,23 +103,29 @@ def test_relationship_simple_link
                 self: '//example.com/link_author/relationships/bio'
               }
             }
-            assert_relationship(:bio, expected)
+
+            author = @author.dup
```
Comment (NullVoxPopuli) [inline test/adapter/json_api/relationships_test.rb]: interesting way to solve the problem.

Does minitest run tests in `//` ? Why isn't `@author` re-set before each of these tests (in setup)? (my lack of minitest knowledge is showing)

**Comment by NullVoxPopuli** [comment]
Can't say I'm a fan of `dup`ing everywhere, as I feel the test framework should be able to handle this. :-\

Do you have thoughts on minitest/spec? `before(:each)`?

File: test/adapter/json_api/relationships_test.rb (hunk 1 above)
Comment (bf4) [inline test/adapter/json_api/relationships_test.rb]: dunno.  my goal was minimal fix, make further changes obvious

**Comment by bf4** [comment]
@NullVoxPopuli at present I prefer using what Rails gives us

**Comment by bf4** [comment]
Any objections to merging or suggestions?  It affected CI while running the throw/catch pr which I want to use in a reflection refactor

**Comment by NullVoxPopuli** [comment]
@bf4 minitest has built in spec syntax

**Pull request merged**